component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/input/Streaming.java

```diff
@@ -34,6 +34,8 @@
 @Slf4j
 public class Streaming {
 
+    public static final int MAX_ACTIVE_TIME_GRACE_PERIOD = 3_000;
```

Comment:
It's just an almost random number. Almost it's like "42".
Less was sometimes not enough, and component-runtime interrupts before the JMS-broker library code can do smth.
Not sure what is the best way how to handle it. 
Balazs suggested some grace period before trying to interrupt. I kind of like it. 
Another idea was is to don't stop by component-runtime if we pass the parameter inside the connector's code.
But it might be a hidden problem when the connector's developer forgets to implement the logic on their side. 
Dunno. I 100% opened for suggestions here